### PR TITLE
[WIP] Service discovery

### PR DESCRIPTION
**Pull request by asridharan**
## Description
DCOS_OSS-836
DCOS_OSS-837

## Urgency
- [ ] Blocker 
- [ ] High
- [* ] Medium

## Requirements
- Test all commands and procedures.
- Build content [locally](https://github.com/dcos/dcos-docs#test-local) and test for formatting/links.
- Add redirects to [dcos-website/redirect-files](https://github.com/dcos/dcos-website#managing-redirects).
- Change all affected versions (e.g. 1.7, 1.8, and 1.9).
- See the [contribution guidelines](https://github.com/dcos/dcos-docs#contributing).

---
This change is [Reviewable](https://reviewable.io/reviews/dcos/dcos-docs/984)

### CODE REVIEW

**Comment by asridharan** [comment]
@joerg84 can you take a look at this?

File: 1.9/overview/networking/load-balacing/index.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+post_title: Load Balancing in DC/OS
+nav_title: Load Balacning
```
Comment (joel-hamill) [inline 1.9/overview/networking/load-balacing/index.md]: typo, should be `Balancing`

File: 1.9/overview/networking/virtual-networks/index.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+post_title: Load Balancing in DC/OS
```
Comment (joel-hamill) [inline 1.9/overview/networking/virtual-networks/index.md]: Probably want a different title here?

File: 1.9/overview/networking/service-discovery/index.md [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+menu_order: 2
+---
+
+DC/OS comes with a highly available, distributed, DNS-based service discovery. There are two main components to the service discovery mechanism in DC/OS. A centralized component called [mesos-dns][4], which runs on every master, and a distributed component called [spartan][6] that runs on every agent. 
+
+Mesos-dns is a centralized, replicated, DNS server that runs on every Master. Every instance of mesos-dns polls the leading Mesos master and generates an FQDN for every service running in DC/OS with the domain `*.mesos`. Since there is a mesos-dns running on each master, this leads to a replicated, highly available DNS service on each of the masters. The high availability of the DNS service is achieved through spartan which acts as a DNS masquerade for mesos-dns on each agent. The spartan on each agent is configured to listen to three different local interfaces on the agent, and the nameservers on the agent is set to these three interfaces. This allows containers to perform upto three re-tries on a DNS request. For each request, spartan forwards it to a different mesos-dns running on each master allowing for a highly available DNS service.
```
Comment (joel-hamill) [inline 1.9/overview/networking/service-discovery/index.md]: You should break this up into a couple of paragraphs

File: 1.9/overview/networking/service-discovery/index.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+
+Apart from acting as a DNS masquerade, to make mesos-dns highly available, the spartan on each agent also acts a DNS server for any service that is load-balanced using the DC/OS internal load balancer called [minuteman][1]. Any service that is load balanced by minuteman gets a [virtual-ip-address (VIP)][1] and an FQDN in the "*.l4lb.thisdcos.directory" domain. The FQDN allocated to a load-balanced service is then stored in spartan. All spartans running in the cluster exchange the records they have learn locally from minuteman using GOSSIP, resulting in a highly available distributed DNS service for any task that is load balanced by minteuman.
```
Comment (joel-hamill) [inline 1.9/overview/networking/service-discovery/index.md]: should be ``*.l4lb.thisdcos.directory``

File: 1.9/overview/networking/service-discovery/index.md [hunk 5]
```diff
@@ -0,0 +1,20 @@
+---
+post_title: Service Discovery in DC/OS
+nav_title: Service Discovery
+menu_order: 2
+---
+
+DC/OS comes with a highly available, distributed, DNS-based service discovery. There are two main components to the service discovery mechanism in DC/OS. A centralized component called [mesos-dns][4], which runs on every master, and a distributed component called [spartan][6] that runs on every agent.
```
Comment (joel-hamill) [inline 1.9/overview/networking/service-discovery/index.md]: Global: spartan should be called the `DNS Forwarder (Spartan)` per the components page. Or at least on its initial occurrence.

**Pull request closed without merging**